Formula/texlive.rb

```diff
@@ -0,0 +1,170 @@
+class Texlive < Formula
+  desc "Free software distribution for the TeX typesetting system"
+  homepage "https://www.tug.org/texlive/"
+  url "https://github.com/TeX-Live/texlive-source/archive/refs/tags/svn58837.tar.gz"
+  sha256 "0afa6919e44675b7afe0fa45344747afef07b6ee98eeb14ff6a2ef78f458fc12"
+  license :public_domain
+  head "https://github.com/TeX-Live/texlive-source.git", branch: "trunk"
+
```

Comment:
> ...and possibly adding a `livecheck` block for this

By default, livecheck will check the Git tags but there are a variety of different tag formats (`build-svn51263`, `pretest-build-20190312`, `pretest-build-svn50430`, `svn58837`, `texlive-2018.2`), so it incorrectly returns `20190312` as the latest version (from the `pretest-build-20190312` tag). This needs a `livecheck` block if we want it to work in a more predictable fashion.

Looking back at older tags, `svn53987` was marked as prerelease, so we can't get away with simply checking the Git tags and matching the ones like `svn58837`. The suggested `livecheck` block below checks the "Latest" release on GitHub and uses a somewhat loose regex due to the variable tag format. [For what it's worth, we only use the `GithubLatest` strategy when it's both correct and necessary and this is one of those cases.]

```suggestion

  livecheck do
    url :stable
    regex(%r{href=["']?[^"' >]*?/tag/\D+(\d+)["' >]}i)
    strategy :github_latest
  end

```

This will work for now and we can always revisit it later (e.g., if/when it breaks).

Reply:
Thanks for the help with this!  As I suspected the variable tag names do complicate things.  We'll have to keep an eye on the releases once TeXLive 2022 comes out next year as that is when I expect new releases to start showing up again.  I'll add this when I push the next set of changes to the PR.

Reply:
I think we can use the updated regex, however after looking at the release tag history, it seems they have switched to the format of `svn#####` after using several alternatives before.  So maybe we actually don't have to worry about capturing all those other tag formats.

@norbusan is the plan to use the `svn#####` format for future tags?  We can certainly change our strategy for identifying new tags if needed, but it's just a bit unclear to us at the moment if there is now a consistent tagging format being used.